### ISSUE TITLE
XAML Islands: ContentDialog not presented in Narrator/NVDA

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#2681]
**Describe the bug**
A ContentDialog is not fully read by accessibility tools when it's opened. This seems to repro when the ContentDialog contains a hyperlink.

**Steps to reproduce the bug**
You will need a XAML Islands app that presents a ContentDialog with a hyperlink in it. For your convenience, you can find one here: [SampleApp.zip](https://github.com/microsoft/microsoft-ui-xaml/files/4783627/SampleApp.zip)
I included the relevant cpp file too.

1. Turn on NVDA/Narrator
2. Run SampleCppApp.exe
3. Invoke either button with spacebar

**Expected behavior**
NVDA/Narrator should read the content dialog and the contents of it.

**Version Info**
| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| June 2020 Update              | Yes   |
| November 2019 Update (18363)  |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**
This is the root cause of https://github.com/microsoft/terminal/issues/5419

**Comment by StephenLPeters** [comment]
@Austin-Lamb and @JesseCol this is an issue from the Terminal team.  @carlos-zamora this is Xaml-Island specific correct?

**Comment by carlos-zamora** [comment]
> @Austin-Lamb Austin Lamb FTE and @JesseCol Jesse Collins FTE this is an issue from the Terminal team. @carlos-zamora Carlos Zamora FTE this is Xaml-Island specific correct?

Yup! The attached SampleApp is just a Xaml Islands app with a Content Dialog. Issue doesn't repro on the Content Dialog in the Xaml Control's gallery.

**Comment by JesseCol** [comment]
Adding @Austin Lamb<mailto:Austin.Lamb@microsoft.com>, @Oscar Calvo<mailto:Oscar.Calvo@microsoft.com>, @Miguel Ramos<mailto:miguelrb@microsoft.com>.

This sounds familiar, do we have a bug in ADO for it somewhere?

Describe the bug
A ContentDialog is not fully read by accessibility tools when it's opened. This seems to repro when the ContentDialog contains a hyperlink.

**Comment by Austin-Lamb** [comment]
@carlos-zamora - I don't expect we'd resolve this until WinUI 3 (or later).  Is this ok for Windows Terminal?

Do you know if this repros with WinUI 3 Preview 2?

**Comment by carlos-zamora** [comment]
@karenbtlai The issue still seems to somewhat happen with NVDA and Narrator. I'm using Windows Terminal's About dialog (using WinUI 2.7 on `main`) as an example.

![Windows Terminal's About dialog](https://user-images.githubusercontent.com/11050425/134237556-1e9a0f77-cd2a-4b70-b875-28563b55955a.png)

The first time you open the about dialog (<kbd>ctrl+shift+space</kbd> to open the new tab flyout, then select "About"), Narrator/NVDA work as expected. They read that there's an About dialog, the focused hyperlink button, then read through the entire content of the dialog.

Surprisingly, any time you open it after that, NVDA/Narrator will only read "documentation link". The dialog is not presented or read in full. I get a consistent repro with NVDA, whereas Narrator happens most of the time (not sure why it doesn't sometimes haha).
